students/john_navitsky/session06/mailroom.py

```diff
@@ -0,0 +1,319 @@
+#!/usr/bin/env python
+
+import sys
+import json
+
+""" Program to manage donations. """
+
+def load_donor_file(donor_file="donors.json"):
+    """ load donors from file into dict, donors """
+    try:
+        with open(donor_file,'r') as donor_data:
+            try:
+                donors = json.load(donor_data)
+            except (json.decoder.JSONDecodeError) as e:
+                # catch malformed json
+                print("The donors file is corrupt!")
+                print(e)
+                print("Please correct or delete the file.")
+                sys.exit(1)
+            return donors
+    except (FileNotFoundError):
+            # if the file isn't found, that's ok, give them
+            # an empty donor list and we'll save anything
+            # they enter upon exit.
+            return []
+    except (PermissionError):
+        # if the file is there, but you can't read it, don't continue
+        # because you won't be able to save the file on exit, risking
+        # loss of data.  Fail early, fail often.
+        print("Insufficent permission to access the donor file!")
+        print("Please correct the permissions.")
+        sys.exit(1)
+
+
+def save_donor_file(donors,donor_file="donors.json"):
+    """ save donors to json file """
+    try:
+        with open(donor_file,'w') as donor_data:
+            donor_data.write(json.dumps(donors))
+        return True
+    except (PermissionError,OSError) as e:
+        print("Sorry, couldn't write the donor file!")
+        print(e)
+        return False
+
+
+def safe_input(prompt=">",mock=False,mock_in=""):
+    """ Generic input routine. """
+    # return null if anything goes wrong
+    if not mock:
+        # normal input mode
+        try:
+            selection=input(prompt).strip()
+        except (KeyboardInterrupt, EOFError):
+            # don't exit the program on ctrl-c, ctrl-d
+            selection=""
+        return selection
+    else:
+        # return what the test program told you to
+        return mock_in
+
+
+def print_lines(lines=2,dest=sys.stdout):
```

Comment:
if you are going to make the output destination settable (which is a great idea) then you may jsut want to go with file writing methods, rather than print() (sys.stdout is a "file-like" object.

```
for i in range(lines):
    dest.write("\n")
```
and when you look at it that way:

`dest.write(lines * "\n")`

:-)